Cython/Compiler/ModuleNode.py

```diff
@@ -3255,17 +3257,17 @@ def generate_module_creation_code(self, env, code):
                 env.module_cname))
         code.putln(code.error_goto_if_null(env.module_cname, self.pos))
         code.putln("#elif CYTHON_COMPILING_IN_LIMITED_API")
-        module_temp = code.funcstate.allocate_temp(py_object_type, manage_ref=True)
+        code.putln("{")
+        module_temp = "module"
         code.putln(
-            "%s = PyModule_Create(&%s); %s" % (
+            "PyObject *%s = PyModule_Create(&%s); %s" % (
                 module_temp,
                 Naming.pymoduledef_cname,
                 code.error_goto_if_null(module_temp, self.pos)))
-        code.put_gotref(module_temp, py_object_type)
         code.putln(code.error_goto_if_neg("PyState_AddModule(%s, &%s)" % (
             module_temp, Naming.pymoduledef_cname), self.pos))
```

Comment:
This leaks a reference to the module now if it fails. That's pretty much why we use managed references also for temps. Or the global module reference.